core/broker/server.go

```diff
@@ -57,8 +56,8 @@ func (b *brokerRPC) Associate(stream pb.Broker_AssociateServer) error {
 		if err != nil {
 			return err
 		}
-		if !uplink.Validate() {
-			return grpcErrf(codes.InvalidArgument, "Invalid Uplink")
+		if err := uplink.Validate(); err != nil {
+			return errors.BuildGRPCError(errors.Wrap(err, "Invalid Uplink"))
```

Comment:
We don't need to wrap here, right? Just use 
`errors.BuildGRPCError(errors.NewErrInvalidArgument("Uplink", err.Error()))`